src/elements/DropdownSelect.tsx

```diff
@@ -75,23 +73,41 @@ const DropdownSelect: FunctionComponent<SelectProps> = ({
   } = useAutocomplete({
     getOptionLabel: (option) => option.name,
     id: "use-autocomplete",
-    options: options,
+    options,
   });
 
   const hasResults = groupedOptions.length > 0;
   const showNoResults = !hasResults && popupOpen;
 
-  // This prevents a form submission when input text does not match any options
+  const { onMouseDown, ...inputProps } = getInputProps();
+
+  /**
+   * This prevents a form submission when input
+   * text does not match any options.
+   */
   const preventFormSubmit = (event: KeyboardEvent): void => {
     if (event.key === "Enter" && inputValue !== value?.name)
       event.preventDefault();
   };
 
+  /**
+   * Disables autocomplete click handler if input is disabled.
+   */
+  const handleMouseDown = (event: MouseEvent<HTMLInputElement>) => {
```

Comment:
I think that fixing the ref issue would solve this problem, so this might not be needed. I tested it on my end with a native input and it marked it as disabled, that's why i didn't add logic for the disabled state

Reply:
Good call, it looks like the MUI `useInput` wasn't respecting the disabled prop.